e2e/testsuite/query/grpc_query.go

```diff
@@ -18,6 +18,11 @@ func GRPCQuery[T any](ctx context.Context, chain ibc.Chain, req proto.Message, o
 		return nil, err
 	}
 
+	return GRPCQueryWithMethod[T](ctx, chain, req, path, opts...)
+}
+
+// GRPCQueryWithMethod queries the chain with a query request with a specific method (grpc path) and deserializes the response to T
+func GRPCQueryWithMethod[T any](ctx context.Context, chain ibc.Chain, req proto.Message, method string, opts ...grpc.CallOption) (*T, error) {
```

Comment:
I had to add this, because the v2 query path is not standard and I think it is better to send in the path rather than have a bunch of ifs here.